engine/table/src/main/java/io/deephaven/engine/table/impl/perf/QueryPerformanceNugget.java

```diff
@@ -279,14 +301,18 @@ private boolean close(final QueryState closingState, final QueryPerformanceRecor
                 return false;
             }
 
+            if (startClockEpochNanos == NULL_LONG) {
+                throw new IllegalStateException("Nugget was never started");
+            }
+
             onBaseEntryEnd();
             endClockEpochNanos = DateTimeUtils.millisToNanos(System.currentTimeMillis());
 
             final RuntimeMemory runtimeMemory = RuntimeMemory.getInstance();
             runtimeMemory.read(endMemorySample);
 
             state = closingState;
-            return recorderToNotify.releaseNugget(this);
+            return onCloseCallback.test(this);
```

Comment:
This is weird. Couldn't we just have the recorder check *after*? Or inside the callback? That way done and abort stop returning a boolean, too.